src/pages/s/[classID]/teachers.tsx

```diff
@@ -0,0 +1,362 @@
+              },
+              {
+                id: 4,
+                name: {
+                  "en-US": "ธนกร_อรรจนาวัฒน์xd",
+                  th: "ธนกร_อรรจนาวัฒน์xd",
+                },
+                via: "line",
+                value: "",
+              },
+              {
+                id: 5,
+                name: {
+                  "en-US": "Thanacord",
+                  th: "Thanacord",
+                },
+                via: "discord",
+                value: "",
+              },
+            ],
+          },
+        },
+      ],
+    },
+  ];
+
+  const [showMain, setShowMain] = useState(teacherList[0].content[0].content);
+
+  showMain.contacts.sort(function (a, b) {
+    if (a.via < b.via) {
+      return -1;
+    }
+    if (a.via > b.via) {
+      return 1;
+    }
+
+    // names must be equal
+    return 0;
+  });
+
+  return (
+    <ListLayout
+      show={true}
+      Title={
+        <Title
+          name={{ title: t("title") }}
+          pageIcon={<MaterialIcon icon="school" />}
+          backGoesTo="/account/login"
+          LinkElement={Link}
+        />
+      }
+    >
+      <ListSection>
+        <CardList
+          listGroups={teacherList}
+          ListItem={({ content, className, onClick, id }) => {
+            return (
+              <button
+                onClick={() => {
+                  onClick();
+                  setShowMain(content);
+                }}
+                className="!w-full"
+              >
+                <TeacherCard
+                  key={content.id}
+                  teacher={content}
+                  hasSubjectSubgroup
+                  className={className}
+                />
+              </button>
+            );
+          }}
+          onChange={() => {}}
+        />
+      </ListSection>
+      <MainSection>
+        <Section className="!flex !flex-col !gap-4 !font-display">
+          <Section>
+            <div className="!sm:gap-6 !md:grid-cols-[1fr_5fr] grid !grid-cols-[1fr_3fr] items-stretch !gap-4">
+              <div className="aspect-square overflow-hidden rounded-xl sm:rounded-2xl">
+                <ProfilePicture src={showMain.profile} />
+              </div>
+              <div className="!flex !flex-col !justify-between">
+                <div>
+                  <h2 className="text-4xl font-bold">
+                    {nameJoiner(locale, showMain.name)}
+                  </h2>
+                  <p className="text-2xl font-medium">
+                    {showMain.subjectsInCharge[0].subjectSubgroup.name[locale]}
+                  </p>
+                </div>
+                <div>
+                  {showMain.classAdvisorAt ? (
+                    <p className="text-2xl font-medium">
+                      {t("advisor", {
+                        className: showMain.classAdvisorAt.name[locale],
+                      })}
+                    </p>
+                  ) : (
+                    <></>
+                  )}
```

Comment:
This can be shortened:
```jsx
{showMain.classAdvisorAt && (
  <p className="text-2xl font-medium">
    {t("advisor", {
      className: showMain.classAdvisorAt.name[locale],
    })}
  </p>
)}
```
This works because of logic. If the first thing in an “and” is false, we know the result is already false and the second thing won’t be considered.